shop/src/main/java/ro/msg/learning/shop/repository/StockRepository.java

```diff
@@ -0,0 +1,23 @@
+package ro.msg.learning.shop.repository;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+import ro.msg.learning.shop.domain.Stock;
+import ro.msg.learning.shop.domain.key.StockKey;
+
+import java.util.UUID;
+
+public interface StockRepository extends JpaRepository<Stock, StockKey> {
+
+    @Query(value = "SELECT * FROM Stock s WHERE s.product_id=?1 AND s.location_id=?2 ", nativeQuery = true)
+    Stock findByProductAndLocation(@Param("productID") UUID productID, @Param("locationID") UUID locationId);
```

Comment:
No need to write a custom query here: you can use only the method name

Reference: https://docs.spring.io/spring-data/data-commons/docs/1.6.1.RELEASE/reference/html/repositories.html